pdns/pdns_recursor.cc

```diff
@@ -3505,6 +3506,10 @@ try
       MT->makeThread(doCarbonDump, 0);
       last_carbon = g_now.tv_sec;
     }
+    if(!t_id && (g_now.tv_sec - last_lua_maintenance >= luaMaintenanceInterval)) {
```

Comment:
So the `maintenance()` hook will only be called from thread `0`? I think we need it to be called for all threads.

Reply:
Since we have one Lua context per thread, I think we really need to be called for all of them. If one needs to do something in a specific thread only, it can use `getRecursorThreadId()` to know from which thread it has been invoked.